src/utils/hooks/useGetFullApplicationStructure.tsx

```diff
@@ -134,13 +150,21 @@ const useGetFullApplicationStructure = (structure: FullStructure, firstRunValida
       evaluationParameters,
       true
     )
-    const results = await Promise.all([isEditable, isRequired, isVisible])
+    const isValid =
+      shouldProcessValidation || firstRunProcessValidation
+        ? evaluateExpressionWithFallBack(element.validationExpression, evaluationParameters, false)
+        : new Promise(() => responseObject[element.code]?.isValid)
```

Comment:
Btw pretty neat with resolving promise after `:`

Reply:
Yeah, as I mentioned above -- it's just a straight value, but wrapped in a Promise just so `isValid` gets handled the same whether or not it gets evaluated/